public/docs/_examples/structural-directives/dart/lib/heavy_loader_component.dart

```diff
@@ -0,0 +1,37 @@
+// #docregion
+import 'package:angular2/angular2.dart';
+import 'dart:async';
+
+int nextId = 1;
+
+@Component(
+    selector: 'heavy-loader',
+    template: '<span>heavy loader #{{id}} on duty!</span>')
+class HeavyLoaderComponent implements OnInit, OnDestroy {
+  int id = nextId++;
+  @Input() List<String> logs;
+
+  ngOnInit() {
+    // Mock todo: get 10,000 rows of data from the server
+    _log(
+        "heavy-loader ${id} initialized, loading 10,000 rows of data from the server");
+  }
+
+  ngOnDestroy() {
+    // Mock todo: clean-up
+    _log("heavy-loader ${id} destroyed, cleaning up");
+  }
+
+  _log(String msg) {
+    logs.add(msg);
+    _tick();
+  }
+
+  /// Triggers the next round of Angular change detection
+  /// after one turn of the JavaScript cycle
+  /// ensuring display of msg added in onDestroy
+  _tick() {
+    new Future(() {});
```

Comment:
Does this really have an effect, a no-op callback in a `Future` that's not returned? Is this just for the side effect of zone intercepting this for change detection. 
Is this telated to

>  You can. Call `markForCheck` on a change detector ref, and then call LifeCycle.tick.

(a comment in http://victorsavkin.com/post/110170125256/change-detection-in-angular-2)